boards/arm/stm32g071b_disco/doc/index.rst

```diff
@@ -0,0 +1,182 @@
+    - USB with Micro-AB (ST-LINK)
+- Board internal connectors:
+    - 2 x 8-pin GPIOs free pins from microcontroller
+	  (accessible internally when case is removed)
+    - USB Type-C™ test points for main signals
+- Flexible power-supply options: ST-LINK USB VBUS or USB Type-C™ VBUS
+- On-board ST-LINK/V2-1 debugger/programmer with USB enumeration capability:
+  mass storage, Virtual COM port and debug port
+
+.. image:: img/stm32g071b_disco.jpg
+   :width: 500px
+   :height: 450px
+   :align: center
+   :alt: STM32G071B-DISCO
+
+More information about the board can be found at the `STM32G071B-DISCO website`_.
+
+
+More information about STM32G071RB can be found here:
+- `G071RB on www.st.com`_
+- `STM32G071 reference manual`_
+
+
+Supported Features
+==================
+
+The Zephyr stm32g071b_disco board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| MPU       | on-chip    | arm memory protection unit          |
++-----------+------------+-------------------------------------+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| COUNTER   | on-chip    | rtc                                 |
++-----------+------------+-------------------------------------+
+| WATCHDOG  | on-chip    | independent watchdog                |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| ADC       | on-chip    | adc                                 |
++-----------+------------+-------------------------------------+
+| DAC       | on-chip    | dac                                 |
++-----------+------------+-------------------------------------+
```

Comment:
Isn't this table a bit misleading? "_The Zephyr stm32g071b_disco board configuration supports the following hardware features_"

The device tree in `boards/arm/stm32g071b_disco/stm32g071b_disco.dts` only has uart3. and the boards yaml file on supports `uart`, `gpio` and `watchdog`. So these peripherals are technically not supported: I2C, SPI, PWM, ADC, DAC, COUNTER. Not sure about MPU and NVIC.

I suppose an overlay file could create the nodes for these peripherals but I think this documentation is inconsistent with other boards. 🤔

Reply:
Just a follow-up comment... Random board example:

* https://docs.zephyrproject.org/2.5.0/boards/arm/stm32f769i_disco/doc/index.html
* https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/stm32f769i_disco/stm32f769i_disco.yaml

 Notice how the documentation does not mention DAC or ADC. Even though the MCU supports it.

Reply:
Thanks for the clarification. I thought the table was supposed to describe the hardware features supported by Zephyr, not the few peripherals in the .dts file. I'll modify the table so that it only lists the peripherals in the .dts file.